misk/web/tabs/web-actions/src/ducks/webActions.ts

```diff
@@ -791,48 +792,58 @@ const generateFieldTypesMetadata = (
   }
 }
 
+const jsonTypeMetadata = OrderedMap<string, ITypesFieldMetadata>().set(
+  "0",
+  buildTypeFieldMetadata(
+    OrderedSet(),
+    "0",
+    "",
+    false,
+    "",
+    ServerTypes.JSON,
+    null
+  )
+)
+
 export const generateTypesMetadata = (
   action: IWebActionAPI
 ): OrderedMap<string, ITypesFieldMetadata> => {
-  const { dispatchMechanism, requestType, types } = action
+  const { dispatchMechanism, pathPattern, requestType, types } = action
   let typesMetadata = OrderedMap<string, ITypesFieldMetadata>().set(
     "0",
     buildTypeFieldMetadata(OrderedSet(), "0")
   )
   if (requestType && types && get(types, requestType)) {
     const { fields } = get(types, requestType)
-    for (const field in fields) {
-      if (fields.hasOwnProperty(field)) {
-        const id = uniqueId()
-        typesMetadata = typesMetadata.mergeDeep(
-          generateFieldTypesMetadata(
-            fields[field],
-            types,
-            typesMetadata,
-            id,
-            "0"
+    try {
+      for (const field in fields) {
+        if (fields.hasOwnProperty(field)) {
+          const id = uniqueId()
+          typesMetadata = typesMetadata.mergeDeep(
+            generateFieldTypesMetadata(
+              fields[field],
+              types,
+              typesMetadata,
+              id,
+              "0"
+            )
           )
-        )
-        typesMetadata = typesMetadata.setIn(
-          ["0", "idChildren"],
-          typesMetadata.getIn(["0", "idChildren"]).add(id)
-        )
+          typesMetadata = typesMetadata.setIn(
+            ["0", "idChildren"],
+            typesMetadata.getIn(["0", "idChildren"]).add(id)
+          )
+        }
       }
+    } catch (e) {
```

Comment:
should we be catching _any_ exception? Wouldn't this mask bugs?

Reply:
Good catch! I'll try and see if we can narrow it down to only catch the call stack errors!

Reply:
"Good catch" 😏